src/models/dashboard-items.js

```diff
@@ -78,40 +96,42 @@ export function itemsMetadataConstructor(dashboardItemsMetadata) {
     const { dataElements, indicators, antigenCategory } = dashboardItemsMetadata;
     const {
         tablesDataCodes: { vaccinesTable, qsIndicatorsTable },
-        chartsDataCodes: { indicatorChart, coverageChart },
+        chartsDataCodes: { indicatorChart, utilizationRateChart },
     } = dashboardItemsConfig;
 
     const vaccineTableDE = dataMapper(dataElements, vaccinesTable);
     const qsTableDE = dataMapper(dataElements, qsIndicatorsTable);
     const indicatorChartInd = dataMapper(indicators, indicatorChart);
-    const coverageChartInd = dataMapper(indicators, coverageChart);
+    const utilizationRateChartInd = dataMapper(indicators, utilizationRateChart);
 
     const dashboardItemsElements = {
         antigenCategory,
         vaccineTable: vaccineTableDE,
         qsTable: qsTableDE,
         indicatorChart: indicatorChartInd,
-        coverageChart: coverageChartInd,
+        utilizationRateChart: utilizationRateChartInd,
     };
     return dashboardItemsElements;
 }
 
-const indicatorsChart = (
+const chartConstructor = (
     name,
     antigen,
     datasetId,
     organisationUnitsIds,
     organisationUnitsParents,
     period,
     antigenCategory,
-    data
+    data,
+    type,
+    appendCode
 ) => ({
-    name: `${name}-${antigen.name}-${dashboardItemsConfig.appendCodes.indicatorChart}`,
-    code: `${datasetId}-${antigen.id}-${dashboardItemsConfig.appendCodes.indicatorChart}`,
+    name: `${name}-${antigen.name}-${appendCode}`,
```

Comment:
Not that it's wrong, but maybe `[str1, str2, str3].join("-")` is easier to read. It may aslo be abstracted to a helper, `buildCode([str1, str2, str3])`, as there are lots of those constructions.

Reply:
agreed, on both this and the underscore